docs/design/2022-06-15-phyical_plan_explain.md

```diff
@@ -0,0 +1,671 @@
+
+- outputting multiple filters which should not show together at the same time.
+
+- unclear representation of the execution process.
+
+- obscure push-down conditions.
+
+The display of the physical plan needs to be improved.
+
+## Background
+
+### The Process of Scan
+
+#### `IndexScan`
+
+In TiSpark, an `IndexScan` requires two scans. One is scanning in index data to get the `RowID` and the other one is scanning in table data to get the real data. The first scan in code is called `IndexScan`, and the second scan in code called `TableScan`, but to facilitate the distinction, in the latter part of the text, the first scan is called `IndexRangeScan`, and the second scan is called `TableRowIDScan`.
+
+#### `CoveringIndexScan`
+
+`CoveringIndexScan` is a special case of `IndexScan`. If the column(s) in Filter and the column(s) in Projection are both inside the Index in one visit, then we only need to scan the Index once, no need to scan for the Table. That is, it only scans for the Index, such a scan is called `CoveringIndexScan`.
+
+#### `TableScan`
+
+`TableScan` is different from `IndexScan` and `CoveringIndexScan`. `TableScan` only scans table data. such a scan we called `TableRangeScan` in the following.
+
+### Filter Concept
+
+#### `PushDown Filter`
+
+The expression passed to COP([TiKV Coprocessor](https://docs.pingcap.com/tidb/v3.0/tikv-overview#tikv-coprocessor) )/TiKV as Selection without triggering a downgrade.
+
+#### `Downgrade Filter`
+
+`Downgrade Filter` is used after `IndexScan` downgrade.
+
+The first stage of `IndexScan` which we called `IndexRangeScan` will return the `RowID` that meets the conditions, and then TiSpark will sort and aggregate the returned Row IDs to obtain the region that needs to be scanned in the second stage of `IndexScan`——`TableRowIDScan`. After sorting and aggregating we will get the `regionTask` number (The number of `regionTask` is equal to the number of ranges to be scanned by `TableRowIDScan`. If the `RowID` returned in the first stage are 1,3,4,5 and 1,3,4 are in the same region and 5 is in another region. Since 1 and 3,4 are not contiguous, 1 is `regionTask`, and since 3,4 and 5 are not in a region, 3,4 is a `regionTask` and 5 is another `regionTask`. ). If the `regionTask` number is bigger than `downgradeThreshold`, The `TableRowIDScan` in the second will be `TableScan`. When `Downgrade Filter` is triggered, the `TableRowIDScan` of the second stage of `IndexScan` will change to `TableRangeScan`. The `StartKey` of `TableRangeScan` is the minimum `RowID` returned by `IndexRangeScan`, and the `EndKey ` of `TableRangeScan` is the minimum `RowID` returned by `IndexRangeScan`. The Filter will become `Downgrade Filter` (`Downgrade Filter` is the same as if the execution plan is `TableScan`'s Filter).
```

Comment:
```suggestion
The first stage of `IndexScan` which we called `IndexRangeScan` will return the `RowID` that meets the conditions, and then TiSpark will sort and aggregate the returned Row IDs to obtain the region that needs to be scanned in the second stage of `IndexScan`——`TableRowIDScan`. After sorting and aggregating we will get the `regionTask` number (The number of `regionTask` is equal to the number of regions to be scanned by `TableRowIDScan`. If the `RowID` returned in the first stage are 1,3,4,5 and 1,3,4 are in the same region and 5 is in another region. Since 1 and 3,4 are not contiguous, 1 is `regionTask`, and since 3,4 and 5 are not in a region, 3,4 is a `regionTask` and 5 is another `regionTask`. ). If the `regionTask` number is bigger than `downgradeThreshold`, the `TableRowIDScan` in the second stage will be `TableScan`. When `Downgrade Filter` is triggered, the `TableRowIDScan` of the second stage of `IndexScan` will change to `TableRangeScan`. The `StartKey` of `TableRangeScan` is the minimum `RowID` returned by `IndexRangeScan`, and the `EndKey ` of `TableRangeScan` is the minimum `RowID` returned by `IndexRangeScan`. The Filter will become `Downgrade Filter` (`Downgrade Filter` is the same as if the execution plan is `TableScan`'s Filter).
```